onnxruntime/core/session/provider_registration.cc

```diff
@@ -150,11 +150,7 @@ ORT_API_STATUS_IMPL(OrtApis::SessionOptionsAppendExecutionProvider,
     status = create_not_supported_status();
 #endif
   } else if (strcmp(provider_name, "VitisAI") == 0) {
-#if defined(USE_VITISAI)
-    options->provider_factories.push_back(VitisAIProviderFactoryCreator::Create(provider_options));
-#else
-    status = create_not_supported_status();
-#endif
+    OrtApis::SessionOptionsAppendExecutionProvider_VitisAI(options, &provider_options);
```

Comment:
why it's not like the other EP's in this file?
e.g. options->provider_factories.push_back(VitisAIProviderFactoryCreator::Create(&provider_options))